EndlessClient/Dialogs/GuildDialog.cs

```diff
@@ -63,11 +78,19 @@ private enum GuildDialogState
             Title = _localizedStringFinder.GetString(EOResourceID.GUILD_GUILD_MASTER);
         }
 
-        private void SetState(GuildDialogState state)
+        protected override void OnUpdateControl(GameTime gameTime)
         {
-            if (state != GuildDialogState.Initial && _state == state)
-                return;
+            if (_guildDescription != _guildSessionProvider.GuildDescription && _guildSessionProvider.GuildDescription != "")
+            {
+                _guildDescription = _guildSessionProvider.GuildDescription;
+                SetState(GuildDialogState.Modify);
+            }
 
+            base.OnUpdateControl(gameTime);
+        }
+
+        private void SetState(GuildDialogState state)
+        {
             _state = state;
```

Comment:
I think you unintentionally removed the `if (_state == state)` early return check here. Definitely don't want/need to be doing all that work if the state is already set correctly.

Reply:
It's how I got the description to re-render. Is there a better way?

Reply:
What I'd probably do is track a reference to the guild description list item (in this PR it's line 257-260) as a member of this dialog and update the text of that list item directly when the guild description changes, rather than having a separate local string in this dialog and then needing to call additional code to actually refresh the string that's really displayed. Because this dialog has multiple states, you'll have to be smart about cleaning up the reference to that list item when the state changes and it's no longer valid; `Optional<T>` is your friend here.

I'll see if I can figure out how to code up a suggestion for the PR since I think that's something that's built in to github's PR flow

Reply:
Now tracking state/buttons/listitemtype per state and using stack


Now tracking the list item in a private optional field as suggested